ci/ci/build.py

```diff
@@ -112,7 +112,8 @@ def visit_dependent(step: Step):
                         raise BuildConfigurationError(f'Step {step.name} cannot be run in cloud {CLOUD}')
                     visited.add(step)
                     for s2 in step.deps:
-                        visit_dependent(s2)
+                        if not s2.run_if_requested:
+                            visit_dependent(s2)
```

Comment:
Hmm, I think this is right because if it is requested, we'll see it in the requeted_step_names loop.

I think we now have duplicate logic though. Check out build.py:159. `name_step[k]` previously had three possible states:
1. key not present, which means invalid step name
2. key mapped to None, which means step is skipped for some reason
3. key mapping to Step, which means step is active

Let's get rid of the if statement from 159, if it is no longer necessary.

Reply:
hmph, I'm starting to see now why `name_step` used to be this value-optional dictionary. I still don't love that the logic is non-local, I'll try to rework it.